File: caesar/caesar.c
#include <cs50.h>
#include <ctype.h>
#include <stdio.h>
#include <string.h>

void cesar(string plaintext, int key);

int main(int argc, string argv[])
{
    // promt: single commandline atg for non neg int and its = k
    if (argc != 2)
    {
        printf("need 1 command-line argument -> number to rotate text by\n");
        return 1;
    }

    // formats the input into a integer
    int key = atoi(argv[1]);

    // prompts for input
    printf("plaintext: ");

    string plaintext = get_string();
    cesar(plaintext, key);


    return 0;
}

// this function converts all the plaintext into caesar cyphertext
void cesar(string plaintext, int key)
{
    int cap_ascii = 65;
    int low_ascii = 97;
    int wrapper = 26;
    printf("cyphertext: ");

    if (plaintext != NULL)
    {
        for (int i=0, n=strlen(plaintext); i<n; i++)
        {
            if (isupper(plaintext[i]))
            {
                printf("%c", ((plaintext[i]-cap_ascii+key)%wrapper)+cap_ascii);
            }
            else if (islower(plaintext[i]))
            {
                printf("%c", ((plaintext[i]-low_ascii+ key)%wrapper)+low_ascii);
            }
            else
            {
                printf("%c", plaintext[i]); // doesn't affect non-letters
            }
        }
        printf("\n");

    }

}

File: caesar/caesar.py
import cs50
import sys

# this function converts all the plaintext into caesar cyphertext
def caesar(plaintext, key):
    # variables
    CAP_ASCII = 65 # capital ascii characters
    LOW_ASCII = 97 # lowercase ascii characters
    WRAPPER = 26 # there are 24 letters in english alphabet
    cyphertext = []
    for i in plaintext:
        if ord(i)>=65 and ord(i)<(65+25): # accepts uppercase
            cypher_char =  chr( ((ord(i)-CAP_ASCII+key) % WRAPPER )+ CAP_ASCII)
            cyphertext.append(cypher_char)
        elif ord(i)>=97 and ord(i)<(97+25): # accepts lowercase
            cypher_char =  chr( ((ord(i) - LOW_ASCII + key) % WRAPPER ) + LOW_ASCII )
            cyphertext.append(cypher_char)
        else:
            cyphertext.append(i)
    return cyphertext

def main():
    # arguments are passed here
    if len(sys.argv) != 2:
        print("arguments needed: key")
        exit(1)
    key = int(sys.argv[1])

    # prompts user for plaintext
    print("Plaintext: ", end=" ")
    plaintext = cs50.get_string()

    # applies the caesar cypher on it
    caesar_cypher = caesar(plaintext, key)

    # formats the cyphertext and gives to user
    for i in caesar_cypher:
        print(i, end='')
    print()
    exit(0)

if __name__ == "__main__":
    main()